fil_actor_interface/src/macros.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019-2022 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+/// `parse_pending_transactions` is a macro for parsing pending transactions and populating a vector with transaction data.
```

Comment:
This does a bit too much, several optional arguments make it difficult to analyze. Did you consider splitting it into smaller methods?

Reply:
yes initially i tried to have it behind one macro only, but looking it now the no of args does seems a lot to manage. 
I think having smaller methods here is a better way to go

Reply:
Let's split it, then.